### DISCUSSION TITLE
[Idea] Snap package for arm64 to support easier installation on Raspberry Pi ?

### DISCUSSION THREAD

**Discussion by SteadEXE** [glpi-project/glpi-agent#333] [Ideas]
### Your idea

Hi,

I don't know how hard it would be to build snap package for arm64.
I know the perl installer already exist but it may be easier for update automation to use a snap package.

At the office I have some Raspberry I can use to test if needed.

It's just an idea

**Comment by g-bougard** [comment]
Hi @SteadEXE 
actually, we are building snap package with GH Actions on Github provided images. And there's not any arm64 images as far I know, or maybe this is possible on MacOSX Silicon images. I don't really know.
Anyway, this could be really easy to create as we are just using the standard snap build process. It's a matter of installing snapcraft, locally export the glpi-agent project and run `snapcraft build` from the sources. If you want to give it a try, you still can read [Snapcraft documentation](https://snapcraft.io/docs). If you find some trouble, you may just have to tune the `snap/snapcraft.yaml` file (and share your tunning of course ;-)).

**Comment by StevenShearinYr7** [comment]
Thanks for the suggestion! Creating a snap package for Scala on arm64 could definitely be a good option for easier installation and updates on that architecture. I'm not personally involved with the Scala project, but I suggest sharing your idea with the Scala community, possibly through their official communication channels such as the Scala forums, mailing list or GitHub repository. This could spark some interest and maybe someone could step up to create and maintain a Scala snap package for arm64. Additionally, if you have experience with creating snap packages, you could also consider contributing to the community by creating one yourself.

**Reply by g-bougard** [reply]
Hi @StevenShearinYr7 
what do you mean by creating a snap package for Scala on arm64 ?
To me, Scala is a programming language and so if you're talking about creating a snap package to install this language on a arm64 linux based distro... then you're in the wrong place. Here we were talking about glpi-agent packaging and it has nothing to do with Scala language.
